src/tests/Services/CollectCmsDataServiceTest.php

```diff
@@ -0,0 +1,179 @@
+        $this->alma_settings_mock->method('is_enabled')->willReturn(true);
+        $this->alma_settings_mock->display_cart_eligibility = 'yes';
+        $this->alma_settings_mock->display_product_eligibility = 'no';
+        $this->alma_settings_mock->display_in_page = false;
+        $this->alma_settings_mock->debug = "yes";
+        $this->alma_settings_mock->method('is_plan_enabled')->willReturn(true);
+        $this->alma_settings_mock->method('get_min_amount')->willReturn(0);
+        $this->alma_settings_mock->method('get_max_amount')->willReturn(1000);
+        $this->fee_plan_mock = $this->createMock(FeePlan::class);
+        $this->fee_plan_mock->method('getPlanKey')->willReturn('general_1_0_0');
+        $this->alma_settings_mock->allowed_fee_plans = [$this->fee_plan_mock];
+        $this->alma_settings_mock->fee_plan_helper = $this->createMock(FeePlanHelper::class);
+        $this->alma_settings_mock->excluded_products_list = [];
+        $this->security_helper_mock = $this->createMock(SecurityHelper::class);
+        $this->payload_formatter_mock = $this->createMock(PayloadFormatter::class);
+        $this->option_proxy_mock = $this->createMock(OptionProxy::class);
+        $valueMap = [
+            ['active_plugins', false, []],
+            ['woocommerce_version', false, 'v.4.2'],
+        ];
+        $this->option_proxy_mock->method('get_option')->willReturnMap($valueMap);
+        $this->theme_proxy_mock = $this->createMock(ThemeProxy::class);
+        $this->theme_proxy_mock->method('get_name')->willReturn('Storefront');
+        $this->theme_proxy_mock->method('get_version')->willReturn('v.4.5');
+        $this->functions_proxy_mock = $this->createMock(FunctionsProxy::class);
+        $this->tools_helper_mock = $this->createMock(ToolsHelper::class);
+
+        $this->collect_cms_data_service = new CollectCmsDataService(
+            $this->alma_settings_mock,
+            $this->alma_logger_mock,
+            $this->payload_formatter_mock,
+            $this->security_helper_mock,
+            $this->option_proxy_mock,
+            $this->theme_proxy_mock,
+            $this->functions_proxy_mock,
+            $this->tools_helper_mock
+        );
+    }
+
+    public function test_send_url()
+    {
+        $this->alma_settings_mock->alma_client = $this->createMock(Client::class);
+        $this->alma_settings_mock->alma_client->configuration = $this->createMock(Configuration::class);
+
+        $this->tools_helper_mock->expects($this->once())
+            ->method('url_for_webhook')
+            ->with(CollectCmsDataService::COLLECT_URL)
+            ->willReturn('http://example.com/woocommerce_api_alma_collect_data_url');
+
+        $this->alma_settings_mock->alma_client->configuration->expects($this->once())
+            ->method('sendIntegrationsConfigurationsUrl')
+            ->with('http://example.com/woocommerce_api_alma_collect_data_url');
+
+        assertNull($this->collect_cms_data_service->send_url());
```

Comment:
this assert haven't $this. And I don't think we assert Null in the test, but Void

Reply:
assertNull check the void return for a fonction too. assertVoid does not exist

Reply:
I know assertVoid doesn't exist. But we can just test if a method is used and remove the assertNull if all the case return null WDYT ?

Reply:
The assertNull ensures that in the event of a modification to the initial function, no new behavior involving a return is introduced.